---
 suspend.c |    1 +
 1 file changed, 1 insertion(+)

Index: suspend-0.8.20081103/suspend.c
===================================================================
--- suspend-0.8.20081103.orig/suspend.c
+++ suspend-0.8.20081103/suspend.c
@@ -2501,6 +2501,7 @@ int main(int argc, char *argv[])
 		if (vt_fd == -ENOTTY) {
 			suspend_warning("Unable to switch virtual terminals, "
 					"using the current console.");
+			splash_param = 0;
 		} else {
 			suspend_error("Could not open a virtual terminal.");
 			ret = errno;
