diff -urN suspend.org/Makefile suspend/Makefile
--- suspend.org/Makefile	2007-02-05 09:52:51.000000000 +0100
+++ suspend/Makefile	2007-02-05 09:57:47.387659171 +0100
@@ -25,7 +26,7 @@
 S2RAM_OBJ=vt.o vbetool/x86-common.o vbetool/vbetool.o radeontool.o dmidecode.o
 SWSUSP_OBJ=vt.o md5.o encrypt.o config.o loglevel.o splash.o bootsplash.o 
 
-S2RAM_LD_FLAGS = $(LD_FLAGS) -lpci
+S2RAM_LD_FLAGS = $(LD_FLAGS) -lpci -lz
 SWSUSP_LD_FLAGS = $(LD_FLAGS)
 
 ifeq ($(ARCH), x86_64)
@@ -81,7 +82,7 @@
 
 #### Rules for objects
 vbetool/x86emu/libx86emu.a:
-	make -C vbetool/x86emu -f makefile.linux
+	$(MAKE) -C vbetool/x86emu -f makefile.linux
 
 s2ram-both.o: s2ram.c s2ram.h whitelist.c
 	$(CC) $(CC_FLAGS) -DCONFIG_BOTH -c $< -o $@
