diff -urN suspend-utils-1.0.g668c5f7.org/configure.ac suspend-utils-1.0.g668c5f7/configure.ac
--- suspend-utils-1.0.g668c5f7.org/configure.ac	2012-09-15 22:23:44.000000000 +0200
+++ suspend-utils-1.0.g668c5f7/configure.ac	2013-07-02 22:05:30.884207944 +0200
@@ -9,7 +9,7 @@
 AC_PREREQ([2.59])
 AC_INIT([suspend-utils], [1.0])
 AM_INIT_AUTOMAKE([${PACKAGE_NAME}], [${PACKAGE_VERSION}])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_SRCDIR([suspend.c])
 AC_CONFIG_MACRO_DIR([m4])
 
diff -urN suspend-utils-1.0.org/bootsplash.c suspend-utils-1.0/bootsplash.c
--- suspend-utils-1.0.org/bootsplash.c	2011-03-07 00:24:15.000000000 +0100
+++ suspend-utils-1.0/bootsplash.c	2013-07-02 22:14:48.196673333 +0200
@@ -118,7 +118,7 @@
 {
 	int ret;
 	bootsplash_to_verbose();
-	printf(prompt);
+	printf("%s", prompt);
 	ret = getchar();
 	bootsplash_to_silent();
 
diff -urN suspend-utils-1.0.org/splash.c suspend-utils-1.0/splash.c
--- suspend-utils-1.0.org/splash.c	2011-03-07 00:24:15.000000000 +0100
+++ suspend-utils-1.0/splash.c	2013-07-02 22:14:40.543075949 +0200
@@ -53,7 +53,7 @@
 
 static int splash_dialog(const char *prompt) 
 {
-	printf(prompt);
+	printf("%s", prompt);
 	return getchar();
 }
 
