--- suspend/radeontool.c	2007-09-13 21:38:10.000000000 +0200
+++ suspend-diet/radeontool.c	2009-04-06 15:09:50.000000000 +0200
@@ -26,6 +26,9 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <sys/mman.h>
+#ifdef __dietlibc__
+#define PCI_HAVE_STDINT_H
+#endif
 #include <pci/pci.h>
 
 #define RADEON_LVDS_GEN_CNTL                0x02d0
--- suspend/s2ram-x86.c	2008-11-03 16:12:53.000000000 +0100
+++ suspend-diet/s2ram-x86.c	2009-04-06 15:09:34.000000000 +0200
@@ -15,6 +15,9 @@
 #include <dirent.h>
 #include <unistd.h>
 
+#ifdef __dietlibc__
+#define PCI_HAVE_STDINT_H
+#endif
 #include <pci/pci.h>
 
 #include "vbetool/vbetool.h"
--- suspend-utils-1.0.g668c5f7/swsusp.h	2013-03-03 02:47:03.715975438 +0200
+++ suspend-utils-1.0.g668c5f7/swsusp.h	2013-03-03 02:47:03.715975438 +0200
@@ -9,6 +9,10 @@
  *
  */
 
+#ifdef __dietlibc__
+typedef uint32_t __kernel_loff_t;
+#endif
+
 #include <stdint.h>
 #include <linux/fs.h>
 #include <linux/suspend_ioctls.h>
@@ -115,6 +119,9 @@
 	return ioctl(dev, SNAPSHOT_PMOPS, PMOPS_FINISH);
 }
 
+#ifdef __dietlibc__
+#define SYS_reboot SYS_REBOOT_NR
+#endif
 static inline void reboot(void)
 {
 	syscall(SYS_reboot, LINUX_REBOOT_MAGIC1, LINUX_REBOOT_MAGIC2,
--- suspend/vbetool/vbetool.c	2008-04-23 17:45:56.000000000 +0200
+++ suspend-diet/vbetool/vbetool.c	2009-04-06 15:10:05.000000000 +0200
@@ -8,6 +8,9 @@
 version 2
 */
 
+#ifdef __dietlibc__
+#define PCI_HAVE_STDINT_H
+#endif
 #include <pci/pci.h>
 #include <assert.h>
 #include <stdio.h>
--- suspend/load.c~	2009-04-06 15:36:54.000000000 +0200
+++ suspend/load.c	2009-04-06 15:43:48.000000000 +0200
@@ -135,10 +135,12 @@
 	memset(handle->extents + n, 0, sizeof(struct extent));
 	handle->cur_extent = handle->extents;
 	handle->cur_offset = handle->cur_extent->start;
+#ifndef __dietlibc__
 	if (posix_fadvise(handle->fd, handle->cur_offset,
 			handle->cur_extent->end - handle->cur_offset,
 			POSIX_FADV_NOREUSE))
 		perror("posix_fadvise");
+#endif
 	return 0;
 }
 
@@ -221,10 +223,12 @@
 	handle->cur_extent++;
 	if (handle->cur_extent->start < handle->cur_extent->end) {
 		handle->cur_offset = handle->cur_extent->start;
+#ifndef __dietlibc__
 		if (posix_fadvise(handle->fd, handle->cur_offset,
 				handle->cur_extent->end - handle->cur_offset,
 				POSIX_FADV_NOREUSE))
 			perror("posix_fadvise");
+#endif
 		return;
 	}
 	/* No more extents.  Load the next extents page. */
--- suspend/resume.c~	2009-01-27 10:48:49.000000000 +0100
+++ suspend/resume.c	2009-04-06 16:53:52.000000000 +0200
@@ -35,6 +35,10 @@
 #include "splash.h"
 #include "loglevel.h"
 
+#if defined(__dietlibc__) && (__WORDSIZE == 64)
+#define lseek64 lseek
+#endif
+
 static char snapshot_dev_name[MAX_STR_LEN] = SNAPSHOT_DEVICE;
 static char resume_dev_name[MAX_STR_LEN] = RESUME_DEVICE;
 static loff_t resume_offset;
