--- Makefile	2006-04-27 23:02:06.782778352 +0200
+++ Makefile-shad	2006-04-27 23:02:20.854639104 +0200
@@ -75,7 +75,7 @@
 	$(CC) -g -O2 -DCONFIG_BOTH -Wall $(CC_FLAGS) md5.o encrypt.o config.o suspend.c s2ram.c -o suspend $(S2RAMOBJ) $(LD_FLAGS) -lpci
 
 resume:	md5.o encrypt.o config.o resume.c swsusp.h config.h encrypt.h md5.h
-	$(CC) -Wall $(CC_FLAGS) md5.o encrypt.o config.o resume.c -static -o resume $(LD_FLAGS)
+	$(CC) -Wall $(CC_FLAGS) md5.o encrypt.o config.o resume.c -o resume $(LD_FLAGS)
 
 ifdef CONFIG_ENCRYPT
 suspend-keygen:	md5.o encrypt.o keygen.c encrypt.h md5.h
