diff -urN suspend-0.5.org/Makefile suspend-0.5/Makefile
--- suspend-0.5.org/Makefile	2006-11-12 15:43:54.000000000 +0100
+++ suspend-0.5/Makefile	2006-11-13 09:25:27.000000000 +0100
@@ -9,6 +9,7 @@
 LD_FLAGS=-L/usr/local/lib
 
 CFLAGS := -O2 -Wall
+LDFLAFS :=
 
 ifdef CONFIG_COMPRESS
 CC_FLAGS	+= -DCONFIG_COMPRESS
@@ -66,7 +67,7 @@
 	rm -f $(S2DISK) $(S2BOTH) swap-offset suspend-keygen suspend.keys resume s2ram *.o vbetool/*.o vbetool/x86emu/*.o vbetool/x86emu/*.a
 
 s2ram:	s2ram.c dmidecode.c whitelist.c radeontool.c $(S2RAMOBJ)
-	$(CC) $(CFLAGS) -g s2ram.c $(S2RAMOBJ) -lpci -o s2ram
+	$(CC) $(CFLAGS) $(LDFLAGS) s2ram.c $(S2RAMOBJ) -lpci -lz -o s2ram
 
 vbetool/vbetool.o:	vbetool/vbetool.c
 	$(CC) $(CFLAGS) -DS2RAM -c vbetool/vbetool.c -o vbetool/vbetool.o
@@ -111,20 +112,20 @@
 	$(CC) -g $(CFLAGS) $(CC_FLAGS) -c $< -o $@
 
 $(S2DISK):	vt.o md5.o encrypt.o config.o suspend.c swsusp.h config.h encrypt.h md5.h $(SPLASHOBJ)
-	$(CC) -g $(CFLAGS) $(CC_FLAGS) vt.o md5.o encrypt.o config.o suspend.c -o $@ $(SPLASHOBJ) $(LD_FLAGS)
+	$(CC) -g $(CFLAGS) $(CC_FLAGS) $(LDFLAGS) vt.o md5.o encrypt.o config.o suspend.c -o $@ $(SPLASHOBJ) $(LD_FLAGS)
 
 $(S2BOTH):	md5.o encrypt.o config.o suspend.c swsusp.h config.h encrypt.h md5.h s2ram.c dmidecode.c whitelist.c radeontool.c $(S2RAMOBJ) $(SPLASHOBJ)
-	$(CC) -g $(CFLAGS) -DCONFIG_BOTH $(CC_FLAGS) md5.o encrypt.o config.o suspend.c s2ram.c -o $@ $(S2RAMOBJ) $(SPLASHOBJ) $(LD_FLAGS) -lpci
+	$(CC) -g $(CFLAGS) -DCONFIG_BOTH $(CC_FLAGS) $(LDFLAGS) md5.o encrypt.o config.o suspend.c s2ram.c -o $@ $(S2RAMOBJ) $(SPLASHOBJ) $(LD_FLAGS) -lpci -lz
 
 resume:	md5.o encrypt.o config.o resume.c swsusp.h config.h encrypt.h md5.h $(SPLASHOBJ)
-	$(CC) $(CFLAGS) $(CC_FLAGS) $(STATIC_CC_FLAGS) md5.o encrypt.o config.o vt.o resume.c $(SPLASHOBJ) -static -o resume $(LD_FLAGS) $(STATIC_LD_FLAGS)
+	$(CC) $(CFLAGS) $(CC_FLAGS) $(STATIC_CC_FLAGS) md5.o encrypt.o config.o vt.o resume.c $(SPLASHOBJ) -static -o resume $(LD_FLAGS) $(STATIC_LD_FLAGS) $(LDFLAGS)
 
 swap-offset: swap-offset.c
-	$(CC) swap-offset.c -o swap-offset
+	$(CC) swap-offset.c -o swap-offset $(LDFLAGS)
 
 ifdef CONFIG_ENCRYPT
 suspend-keygen:	md5.o keygen.c encrypt.h md5.h
-	$(CC) $(CFLAGS) -DHAVE_INTTYPES_H -DHAVE_STDINT_H $(CC_FLAGS) md5.o keygen.c -o suspend-keygen $(LD_FLAGS)
+	$(CC) $(CFLAGS) -DHAVE_INTTYPES_H -DHAVE_STDINT_H $(CC_FLAGS) md5.o keygen.c -o suspend-keygen $(LD_FLAGS) $(LDFLAGS)
 
 install-s2disk: $(S2DISK) suspend-keygen swap-offset conf/$(CONFIGFILE) $(SNAPSHOT)
 	install --mode=755 suspend-keygen $(DESTDIR)$(SUSPEND_DIR)
